### PR TITLE
Drugbank gene drug

### PR DESCRIPTION
**Pull request by jordan2lee**
Add association node. Allows for gene-drug associations from drugbank to be added as a property. 

Connects to existing nodes: gene, compound, publication

### CODE REVIEW

**Comment by kellrott** [comment]
I think `Association` is too generic of a term. Something like `GeneDrugInteraction` might be more appropriate.

**Comment by kellrott** [comment]
Looking at the `links`, there are a bunch of properties that are referenced that are never declared, including `genes`, `compounds` and `publications`

File: gdcdictionary/schemas/gene_drug_association.yaml [hunk 1]
```diff
@@ -0,0 +1,76 @@
+$schema: "http://json-schema.org/draft-04/schema#"
+
+id: "protein_drug_association"
```
Comment (kellrott) [inline gdcdictionary/schemas/gene_drug_association.yaml]: Try to be consistent in naming. Either rename file to be `protein_drug_association` or change class to be `gene_drug_association`

File: gdcdictionary/schemas/gene_drug_association.yaml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  - project_id
+
+uniqueKeys:
+  - [id]
+  - [project_id, submitter_id]
+
+links:
+  - backref: protein_drug_association
+    label: proteins
+    multiplicity: many_to_many
```
Comment (kellrott) [inline gdcdictionary/schemas/gene_drug_association.yaml]: If this is `many_to_many`, then `proteins` needs to be an array

File: gdcdictionary/schemas/gene_drug_association.yaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+properties:
+  association_type:
+    type: string
+    description: "protein-drug association keyword, example inhibitor"
+  compound:
+    type: string
+    description: "Raw compound ID"
+  proteins:
+    type: string
+    description: "Raw proteins"
```
Comment (kellrott) [inline gdcdictionary/schemas/gene_drug_association.yaml]: Plural, but not an array

File: gdcdictionary/schemas/gene_drug_association.yaml [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+  project_id:
+    $ref: _definitions.yaml#/project_id
+  publications:
+    type: string
+    description: "Raw pubmed publication IDs"
```
Comment (kellrott) [inline gdcdictionary/schemas/gene_drug_association.yaml]: Plural, but not an array

**Review by kellrott — CHANGES_REQUESTED** [review]

File: gdcdictionary/schemas/gene_drug_association.yaml (hunk 4 above)
Comment (jordan2lee) [reply gdcdictionary/schemas/gene_drug_association.yaml]: updated in commit 
b4a6400

File: gdcdictionary/schemas/gene_drug_association.yaml (hunk 3 above)
Comment (jordan2lee) [reply gdcdictionary/schemas/gene_drug_association.yaml]: updated in commit b4a6400

File: gdcdictionary/schemas/gene_drug_association.yaml (hunk 2 above)
Comment (jordan2lee) [reply gdcdictionary/schemas/gene_drug_association.yaml]: `one_to_one` in commit b4a6400

**Comment by kellrott** [comment]
For arrays of id strings, many of the other structures use a reference to a data type definition. Example
```
  aliquots:
    $ref: _definitions.yaml#/to_many
```
Try using this for proteins and publications

**Review by kellrott — APPROVED** [review]
LGTM

**Pull request merged**